### PR TITLE
CRWA installation success message

### PR DESCRIPTION
**Pull request by jeliasson**
**Description**
Updates the success message when a Redwood application is created using `yarn create redwood-app`.

**Related**
#1171 Improve final "success" output during CRWA installation

**Questions**
- [X] Would we rather have this change in a series of `console.log`'s (as before) or rendered from an [object](https://github.com/redwoodjs/redwood/pull/1195/commits/7373b0de2374383ddd5d906e0bd67bcba14efc92#diff-bb0a9e2525a05a79a2b81b652eb44480R199) (as of PR)?
- [X] Is it okay to do some refactoring of methods and helpers into separate files? If so, I'll will do this in another PR.
- [X] Is there any lightweight core package that would be suitable in the future to share chalk styles/colors for`@redwood/cli` and `@redwood/create-redwood-app`? (See [this](https://github.com/redwoodjs/redwood/blob/main/packages/cli/src/lib/colors.js) and [this](https://github.com/redwoodjs/redwood/pull/1195/commits/7373b0de2374383ddd5d906e0bd67bcba14efc92#diff-bb0a9e2525a05a79a2b81b652eb44480R188))

**Latest output**
image

### CODE REVIEW

**Comment by thedavidprice** [comment]
fixes #1171

_Continuing conversation from Issue:_
> Would we rather have this change in a series of console.log's (as before) or rendered from an object (as of PR)?

^^ Object looks good to me! 

> Is it okay to do some refactoring of methods and helpers into separate files? If so, I'll do this in another PR.

^^ I'd vote to hold off on a re-org in this PR. I'd rather get it shipped.

> Is there any lightweight core package that would be suitable in the future to share chalk styles/colors for@redwood/cli and @redwood/create-redwood-app?

^^ Ah, that’s 💯 the right way to do it, but I'm unsure how to do this the right way with Lerna (what we use to manage monorepo and publishing packages). Let's see if Peter has a suggestion — if not, leave it as is so we don't let this PR get stuck on optimizing this.

👉 @peterp if create-redwood-app imports `cli/src/lib/colors.js`, is Lerna smart enough to just import the file for build+publish? Or will it effectively require `@redwoodjs/cli`, which is 220kb? And any other suggestions about how to do this?

File: packages/create-redwood-app/README.md [hunk 1]
```diff
@@ -27,16 +31,23 @@ Explains how to contribute by addressing the following three points:
...
+node -r esm ./packages/create-redwood-app/src/create-redwood-app.js --no-yarn-install ../create-redwood-app-test
 
+# For the lazy; forcefully delete ../create-redwood-app-test and then run without installing yarn dependencies
+rm -rf ../create-redwood-app-test && node -r esm ./packages/create-redwood-app/src/create-redwood-app.js --no-yarn-install ../create-redwood-app-test
 ```
-npm install --save esm
-```
\ No newline at end of file
+
+_Note: In above example we are creating the app in the parent directory in folder `create-redwood-app-test`._
```
Comment (peterp) [inline packages/create-redwood-app/README.md]: These instructions could potentially trip you up; I would suggest doing the following instead:

1. Run `yarn build:watch` command which will transpile your code when it's modified.
2. Run `node ./dist/create-redwood-app.js` to test.

File: packages/create-redwood-app/src/create-redwood-app.js [hunk 2]
```diff
@@ -164,6 +164,97 @@ const installNodeModulesTasks = ({ newAppDir }) => {
   ]
 }
 
+const displaySuccessMessage = () => {
+  const items = displaySuccessMessageItems
+
+  return Object.keys(items).forEach((key) => {
```
Comment (peterp) [inline packages/create-redwood-app/src/create-redwood-app.js:170]: Checkout `Object.entries`

**Comment by peterp** [comment]
> @jeliasson Would we rather have this change in a series of console.log's (as before) or rendered from an object (as of PR)?

> @thedavidprice ^^ Object looks good to me!

I would actually suggest just using console.log, the object is nice if we were to re-use it everywhere, but this adds a layer of indirection that is difficult to extend, a bit more complicated to parse; maybe something like this could be an alternative:

```js
[
style.success('Thanks for trying out Redwood!'),
'',
'',
`We've created your app in '${style.cmd(newAppDir)}'`,
`Enter the directory and run '${style.cmd('yarn rw dev')}' to start the development server.`
].map(console.log)
```

versus:

```js
 start: {
    header: style.success('Thanks for trying out Redwood!'),
    rows: [
      {
        text: `We've created your app in '${style.cmd(newAppDir)}'`,
      },
      {
        text: `Enter the directory and run '${style.cmd(
          'yarn rw dev'
        )}' to start the development server.`,
      },
    ],
  },
```

**Comment by peterp** [comment]
> if create-redwood-app imports cli/src/lib/colors.js, is Lerna smart enough to just import the file for build+publish? Or will it effectively require @redwoodjs/cli, which is 220kb? And any other suggestions about how to do this?

@thedavidprice We should keep packages separated. If we want to share colours across CLI commands then I think we could create another package, but I think that might be more complicated than it's worth.

`ansi-colors` is a really great alternative.

File: packages/create-redwood-app/README.md (hunk 1 above)
Comment (jeliasson) [reply packages/create-redwood-app/README.md]: @peterp Much better! I'll update the docs.

File: packages/create-redwood-app/src/create-redwood-app.js (hunk 2 above)
Comment (jeliasson) [reply packages/create-redwood-app/src/create-redwood-app.js:170]: @peterp TIL, thank you 😄 
However, I will make an update to PR using `.map()` as you suggested below. More straight forward in this case.

**Comment by jeliasson** [comment]
> `ansi-colors` is a really great alternative to chalk. I would argue that maybe we shouldn't be using hex colours and should stick to what a colour library gives us, since these are "safe colours" for most terminals.

`.hex()` will [down sample](https://github.com/chalk/chalk#256-and-truecolor-color-support) to ANSI color format that is supported by the terminal (or specified `level` in Chalk options). I'll see if I can make a palette that looks somewhat the same with different levels of down sample. That way we can continue to use `chalk` if we want.

**Comment by peterp** [comment]
@jeliasson Oooh, I didn't know that! TIL!

File: packages/create-redwood-app/src/create-redwood-app.js [hunk 3]
```diff
@@ -169,12 +169,12 @@ const installNodeModulesTasks = ({ newAppDir }) => {
...
   warning: chalk.keyword('orange'),
-  success: chalk.hex('#c3e88d'),
+  success: chalk.greenBright,
   info: chalk.grey,
   header: chalk.bold.underline.hex('##e8e8e8'),
   cmd: chalk.hex('#808080'),
   redwood: chalk.hex('#ff845e'),
-  love: chalk.reset.hex('#de1d10'),
+  love: chalk.redBright,
 }
```
Comment (jeliasson) [inline packages/create-redwood-app/src/create-redwood-app.js:179]: This will look as follows

image

**Review by jeliasson — COMMENTED** [review]
Suggestion to a style palette with it's supported levels of ANSI colors (`1-3` where as `0` is just white).

File: packages/create-redwood-app/src/create-redwood-app.js (hunk 3 above)
Comment (peterp) [reply packages/create-redwood-app/src/create-redwood-app.js:179]: This looks really good <3, learning a ton here :)

**Comment by jeliasson** [comment]
@peterp @thedavidprice 
Please review at your convenience and let me know in case of changes. I'll be happy to refactor parts of `create-redwood-app` in a separate PR down the road.

CC @mojombo @cannikin

**Review by peterp — APPROVED** [review]
Awesome!

**Review by thedavidprice — APPROVED** [review]

**Comment by thedavidprice** [comment]
@jeliasson Ran it locally to see the output and really looks 🤩 Thank you!

Suggestions for Next Steps (no pressure):
- in a new Issue, map out how you'd like to re-structure and cc Peter and me. 
- in an attempt to keep CLI styles consistent, could you add a comment to the `cli/src/lib/colors.js` file that mentions/references styles you created here with a suggestion to keep them consistent ongoing? (Note: you don't need to list all the style in the comment, just add the path to this file as a reference.)

**Pull request merged**